### PR TITLE
testing: remove go-vcr dependency: we no longer use it

### LINKED ISSUE #770: all: use a better replay tool
With the existing replay tool, order of RPCs matters, making concurrent tests unusable. The tool also doesn't have a way to capture state that may vary between runs, like times and random numbers.

Replace with `cloud.google.com/go/httpreplay` and/or `cloud.google.com/go/rpcreplay`.

### PR DESCRIPTION
**Pull request by eliben**
Fixes #770

### CODE REVIEW

**Comment by eliben** [comment]
Reran listdeps to fix travis

File: internal/testing/replay/replay.go [hunk 1]
```diff
@@ -16,168 +16,13 @@
...
-				}
-			}
-		}
-		keep = append(keep, action)
-	}
-	c.Interactions = keep
-	c.Mu.Unlock()
-	return c.Save()
-}
-
```
Comment (vangent) [inline internal/testing/replay/replay.go]: Can we move `NewGCPDialOptions` to `setup.go` and just delete this `replay` package entirely?

Comment (vangent) [reply internal/testing/replay/replay.go]: (feel free to do in a separate PR)

**Comment by vangent** [comment]
\o/ thanks @eliben !!!

**Comment by shantuo** [comment]
Also remove the old recorded files?

File: internal/testing/replay/replay.go (hunk 1 above)
Comment (eliben) [reply internal/testing/replay/replay.go]: Yep, have separate PR in progress for that

**Comment by eliben** [comment]
> Also remove the old recorded files?

Oops, yes. will do in a separate PR

**Pull request merged**